### DISCUSSION TITLE
Preventing macOS firewall modal dialog

### DISCUSSION THREAD

**Discussion by bsergean** [anacrolix/torrent#539] [Q&A]
Hi there,

When I create a client object in a unittest context (which needs to run headless) I would like to avoid this modal dialog that display: `Do you want the application “main” to accept incoming network connections?`.

I am trying to set the ListHost to only localhost, but this isn't enough. (when I try to return 127.0.0.1 things does not work).

```
clientConfig.ListenHost = func(string) string { return "localhost" }
```

Does someone has any clue about this, or uses macOS ?

**Comment by anacrolix** [comment]
I develop on MacOS, I haven't seen this. Do you have firewall enabled?

**Comment by bsergean** [comment]
Yes, my Mac is my office / company one, and they probably have the ’secure’ default settings, where I believe that binding anything out of 0.0.0.0 kicks in the warning.

**Comment by bsergean** [comment]
It isn’t as bad as I thought actually.

If I compile my binary (go build …), and use the compiled binary name which stays stable, as opposed to using go run which will create a different unique/tmp binary name each time, the macOS firewall will remember that my binary is ok to run, after I check ‘Allow’ when the modal dialog pop up.

So on the next run the modal dialog does not show up.

**Reply by anacrolix** [reply]
Ok thanks for the follow-up. I expect this might occur for any Go programs that use the network?
